components/navigable-container/index.js

```diff
@@ -124,19 +124,28 @@ export class NavigableMenu extends Component {
 		const eventToOffset = ( evt ) => {
 			const { keyCode } = evt;
 
-			const isVertical = orientation === 'vertical';
-			const isHorizontal = orientation === 'horizontal';
-
-			// Still handle any arrow keys, even if the opposite orientation
-			if ( LEFT === keyCode ) {
-				return isHorizontal ? -1 : 0;
-			} else if ( UP === keyCode ) {
-				return isVertical ? -1 : 0;
-			} else if ( RIGHT === keyCode ) {
-				return isHorizontal ? +1 : 0;
-			} else if ( DOWN === keyCode ) {
-				return isVertical ? +1 : 0;
-			}
+			const offsetMap = {
+				vertical: {
+					[ LEFT ]: 0,
+					[ RIGHT ]: 0,
+					[ UP ]: -1,
+					[ DOWN ]: +1,
+				},
+				horizontal: {
+					[ LEFT ]: -1,
+					[ RIGHT ]: +1,
+					[ UP ]: 0,
+					[ DOWN ]: 0,
+				},
+				vertical_two_columns: {
+					[ LEFT ]: -1,
+					[ RIGHT ]: +1,
+					[ UP ]: -2,
+					[ DOWN ]: +2,
+				},
+			};
+
+			return offsetMap.hasOwnProperty( orientation ) ? offsetMap[ orientation ][ keyCode ] : 0;
```

Comment:
With the way that it works at the moment, it's important to only return 0 if you want the event stopped. You need to return `undefined` if the event shouldn't be handled. These components should only stop arrow key events, and let everything else through.